### PR TITLE
WIP: run one UFS WM test

### PR DESCRIPTION
**Pull request by rickgrubin-noaa**
### Summary

Add testing for one `UFS WM` test.

### Testing

**Orion**

- compiler: `Intel`
- tool: `Jenkins` pipeline
- test results (must have `epic` role account access):
  - UFS WM run dir:
    - `/work/noaa/epic/role-epic/jenkins/workspace/spack-stack/ufs/Orion/stmp/role-epic/stmp/role-epic/FV3_RT/rt_2528430`
      - `control_c48_intel.log` (single test result)
  - Single test logs:
    - `/work/noaa/epic/role-epic/jenkins/workspace/spack-stack/ufs/Orion/ufs-weather-model/tests/logs`
      - `RegressionTests_orion.log` (regression tests)

### Applications affected

n/a

### Systems affected

n/a

### Dependencies

Utilizes  [UFS WM @ spack-stack-automation](https://github.com/rickgrubin-noaa/ufs-weather-model/tree/spack-stack-automation)

### Issue(s) addressed

Completes [pr/1213](https://github.com/JCSDA/spack-stack/pull/1213)

### Checklist
- [ ] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [ ] These changes have been tested on the affected systems and applications.
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA @rickgrubin-noaa I pulled this into my working branch for Nautilus and it worked fine. Good to merge as far as I am concerned.

File: util/weekly_build/apptests/test_ufswm.sh [hunk 1]
```diff
@@ -3,3 +3,18 @@
...
+cd ${RUNDIR}
+
+# The following need to be cleaned up for proper variable substitution; this is a one-off test
+BRANCH=spack-stack-automation
+
+git clone -b ${BRANCH} --single-branch --recurse-submodules https://github.com/rickgrubin-noaa/ufs-weather-model.git
+cd ufs-weather-model/tests
+
+# -r ==> rocoto ; not strictly necessary, can run without
+./rt.sh -a epic -k -r -n "control_c48 intel"
```
Comment (AlexanderRichert-NOAA) [inline util/weekly_build/apptests/test_ufswm.sh]: This looks great-- Is there a way to generalize this for systems that won't use the "epic" job scheduler account? Maybe a variable that could be set in the site configs?

Comment (rickgrubin-noaa) [reply util/weekly_build/apptests/test_ufswm.sh]: Indeed, generalizations are necessary for anything that may be different across various `tier1` and `tier2` hosts. 

Certainly 

- `branch`
- `job scheduler`
- `job scheduler account` 

are the obvious variables; `rt.sh` has numerous options that could be parameterized in this sense, and should be. If that's preferable prior to merging, that is fine / I'll do that.

Comment (rickgrubin-noaa) [reply util/weekly_build/apptests/test_ufswm.sh]: Scripts cleaned up:

- `06_AppTests.sh`
  - intended for site-specific modification with respect to how `ufs-weather-model/tests/rt.sh` is executed with respect to what arguments are passed to that overarching test script
- `apptests/test_ufswm.sh`
  - `ufs-weather-model/tests/rt.sh` will parse arguments passed through from `06_AppTests.sh`

Tested on **orion** via `Jenkins` and from the command line.

Changes to `ufs-weather-model` are required to test against a stack built via automation, as `ufs-weather-model/tests/rt.sh` expects to use a most recently released / published stack:

- `ufs-weather-model/tests/rt.sh`
  - may need to change the value of `dprefix` to a location where the `$USER` running the automated build has `rwx` privileges
- `ufs-weather-module/modulefiles/ufs_<host>.<compiler>.lua` must be changed to reflect:
  - location of automated build stack modules
  - version(s) of modules loaded, if different from past stack releases
    - e.g. newer compiler

File: util/weekly_build/06_AppTests.sh [hunk 2]
```diff
@@ -2,8 +2,19 @@
...
+for compiler in $COMPILERS; do
+  cd $RUNDIR/$RUNID/envs/build-${compiler/@/-}
+  spack env activate .
+  spack module lmod refresh -y
+  spack stack setup-meta-modules
+done
 
 if [ "$TEST_UFSWM" == ON ]; then
-  ./apptests/test_ufswm.sh
+  ./apptests/test_ufswm.sh $*
```
Comment (AlexanderRichert-NOAA) [inline util/weekly_build/06_AppTests.sh]: Would it be possible to configure the rt.sh arguments so they can be set by an environment variable, kind of like you did with UFSWM_BRANCH and UFSWM_URL? I think that will make both site-specific configuration and debugging/testing easier.

Comment (rickgrubin-noaa) [reply util/weekly_build/06_AppTests.sh]: Site-specific config (for **orion**) added, generalizing `rt.sh` arguments

File: util/weekly_build/apptests/test_ufswm.sh [hunk 3]
```diff
@@ -3,3 +3,19 @@
 # Do some stuff, and exit non-zero if UFSWM cannot be successfully built and
 # tested (ideally, have a discernably different error condition if there are
 # numerical differences)
+
+echo Base directory: ${RUNDIR:?}
+cd ${RUNDIR}
+
+UFSWM_BRANCH=${UFSWM_BRANCH:-develop}
+UFSWM_URL=${UFSWM_URL:-"https://github.com/ufs-community/ufs-weather-model.git"}
```
Comment (AlexanderRichert-NOAA) [inline util/weekly_build/apptests/test_ufswm.sh:11]: As it stands, will ufs-weather-model be able to pick up on the modules in the installation path we're testing? Or is this going to use the MODULEPATH settings currently in the ufs-weather-model repo?

Comment (rickgrubin-noaa) [reply util/weekly_build/apptests/test_ufswm.sh:11]: As noted prior:

Changes to `ufs-weather-model` are required to test against a stack built via automation, as `ufs-weather-model/tests/rt.sh` and associated lua files, in the `develop` branch, expect to use a most recently released / published stack:

- `ufs-weather-model/tests/rt.sh`
  - may need to change the value of `dprefix` to a location where the `$USER` running the automated build has `rwx` privileges
- `ufs-weather-module/modulefiles/ufs_<host>.<compiler>.lua` must be changed to reflect:
  - location of automated build stack modules
  - version(s) of modules loaded, if different from past stack releases
    - e.g. newer compiler

Setting `UFSWM_BRANCH` appropriately in `sites/<site>.sh` allows for this; for example, see:

- [ufs_orion.intel.lua](https://github.com/rickgrubin-noaa/ufs-weather-model/blob/spack-stack-automation/modulefiles/ufs_orion.intel.lua)
- [tests/rt.sh](https://github.com/rickgrubin-noaa/ufs-weather-model/blob/spack-stack-automation/tests/rt.sh#L785)

**Pull request merged**